server/plugins/schema-plugin-ob-oracle/src/main/java/com/oceanbase/odc/plugin/schema/oboracle/OBOracleTableExtension.java

```diff
@@ -122,11 +123,21 @@ private String getTableDDL(Connection connection, String schemaName, String tabl
             }
         }
         List<DBTableIndex> indexes = parser.listIndexes();
+        List<String> constraintNames = parser.listConstraints().stream().map(DBTableConstraint::getName).filter(
+                Objects::nonNull).collect(Collectors.toList());
+        List<DBTableConstraint> constraintNameIsNull =
+                parser.listConstraints().stream().filter(cons -> Objects.isNull(cons.getName())).collect(
+                        Collectors.toList());
         for (DBTableIndex index : indexes) {
             /**
-             * 如果有唯一索引，则在表的 DDL 里已经包含了对应的唯一约束 这里就不需要再去获取索引的 DDL 了，否则会重复
+             * If it is a unique index and the corresponding unique constraint is already included in the DDL of
+             * the table, there is no need to obtain the DDL of the index, otherwise it will be repeated.
              */
-            if (index.getType() == DBIndexType.UNIQUE || index.getPrimary()) {
+            if (index.getPrimary() || constraintNames.contains(index.getName())
+                    || constraintNameIsNull.stream()
+                            .anyMatch(con -> DBConstraintType.UNIQUE_KEY == con.getType()
+                                    && con.getColumnNames().containsAll(index.getColumnNames())
```

Comment:
if an index use same columns as a constraint, but diffrent with column order,
may should not consider as equal ?

may refer Lists.equals

Reply:
You are right～